presto-main/src/main/java/com/facebook/presto/sql/analyzer/Scope.java

```diff
@@ -39,6 +39,7 @@
 public class Scope
 {
     private final Optional<Scope> parent;
+    private final boolean queryBoundary;
```

Comment:
I think it should the implementation of `resolve` method

In case of:
```
- outerQueryScope with [a, b]
  - scope [b]
    - scope [a]
```

When you look for `b` from bottom most scope then I think it should return `b` from the `outerQueryScope`. Unless it is hierarchical scope (introduced later), but in this case all the scope should have queryBoundary always set to `true`, unless they are hierarchical.